src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Definition/RazorDefinitionEndpoint.cs

```diff
@@ -165,29 +237,49 @@ public DefinitionRegistrationOptions GetRegistrationOptions(DefinitionCapability
             if (node is null)
             {
                 logger.LogInformation("Could not locate ancestor of type MarkupTagHelperStartTag or MarkupTagHelperEndTag.");
-                return null;
+                return (null, null);
             }
 
             var name = GetStartOrEndTagName(node);
             if (name is null)
             {
                 logger.LogInformation("Could not retrieve name of start or end tag.");
-                return null;
+                return (null, null);
+            }
+
+            string? propertyName = null;
+
+            // If we're on an attribute then just validate against the attribute name
+            if (owner.Parent is MarkupTagHelperAttributeSyntax attribute)
+            {
+                name = attribute.Name;
+                propertyName = attribute.TagHelperAttributeInfo.Name;
             }
 
             if (!name.Span.Contains(location.AbsoluteIndex))
             {
-                logger.LogInformation($"Tag name's span does not contain location's absolute index ({location.AbsoluteIndex}).");
-                return null;
+                logger.LogInformation($"Tag name or attributes's span does not contain location's absolute index ({location.AbsoluteIndex}).");
+                return (null, null);
             }
 
             if (node.Parent is not MarkupTagHelperElementSyntax tagHelperElement)
             {
                 logger.LogInformation("Parent of start or end tag is not a MarkupTagHelperElement.");
-                return null;
+                return (null, null);
             }
 
-            return tagHelperElement.TagHelperInfo.BindingResult;
+            var originTagDescriptor = tagHelperElement.TagHelperInfo.BindingResult.Descriptors.FirstOrDefault(d => !d.IsAttributeDescriptor());
+            if (originTagDescriptor is null)
+            {
+                logger.LogInformation("Origin TagHelper descriptor is null.");
+                return (null, null);
+            }
+
+            var attributeDescriptor = (propertyName is not null)
+                ? originTagDescriptor.BoundAttributes.FirstOrDefault(a => a.Name.Equals(propertyName, StringComparison.Ordinal))
```

Comment:
IIRC in some Blazor scenarios this can fall over. I forget but does Blazor special case dictionaries @TanayParikh? If Blazor treats them like typical TagHelpers I'd imagine the name for a dictionary object might be something like `SomeKey` but the attribute names will be `SomeKey-AValue` and at runtime the dictionary will be filled with `SomeKey` -> `AValue` entries . I think using [this](https://github.com/dotnet/razor-compiler/blob/52505c754be0bc118a0f335eba10742e36eeb965/src/Microsoft.AspNetCore.Razor.Language/src/TagHelperMatchingConventions.cs#L92) should cover that though.

Reply:
Had a quick look on google and I don't think dictionaries will be a problem, but certainly can follow up.